### PR TITLE
Handle letter with number of document copies specified

### PR DESCRIPTION
**Pull request by rkondratowicz**
Only `PdfCreator` changes here.
`LetterService` and controller changes in next PRs.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/model/in/LetterWithPdfsAndCopiesRequest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import java.util.List;
+import java.util.Map;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Size;
+
+public class LetterWithPdfsAndCopiesRequest implements ILetterRequest {
+
+    @Size(min = 1, max = 30)
+    public List<Doc> documents;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/model/in/LetterWithPdfsAndCopiesRequest.java]: should this be final ?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/sendletter/model/in/LetterWithPdfsAndCopiesRequest.java]: yup. fixed.

**Comment by lgonczar** [comment]
I think the most value in this story comes from the fact that we could avoid dealing with potentially large amounts of data in memory. For that to happen, we need to modify PdfMerger to accept a collection of streams and return a stream, and use it like this in PdfCreator.

**Comment by lgonczar** [comment]
Discussion would have been nice

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/pdf/PdfCreator.java [hunk 2]
```diff
@@ -43,6 +46,19 @@ public PdfCreator(DuplexPreparator duplexPreparator, IHtmlToPdfConverter convert
...
     }
 
+    public byte[] createFromBase64PdfWithCopies(List<Doc> docs) {
+        Asserts.notNull(docs, "base64decodedDocs");
+
+        List<byte[]> pdfs = docs
+            .stream()
+            .map(d -> Collections.nCopies(d.copies, d.content))
+            .flatMap(Collection::stream)
+            .map(duplexPreparator::prepare)
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/pdf/PdfCreator.java]: I think this we should do this before nCopies operations as this method creates new copies ???

**Pull request merged**